### PR TITLE
CLN: clean-up base.py

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Part of https://github.com/geopandas/geopandas/pull/835 that is unrelated to ExtensionArrays, so split off.

Harmonized the helper functions for binary/unary spatial ops + some PEP8 clean-up.

### CODE REVIEW

File: geopandas/base.py [hunk 1]
```diff
@@ -289,13 +316,13 @@ def geom_almost_equals(self, other, decimal=6):
             Decimal place presion used when testing for approximate equality.
         """
         # TODO: pass precision argument
-        return _series_op(self, other, 'almost_equals', decimal=decimal)
+        return _binary_op('almost_equals', self, other, decimal=decimal)
 
     def geom_equals_exact(self, other, tolerance):
         """Return True for all geometries that equal *other* to a given
         tolerance, else False"""
         # TODO: pass tolerance argument.
```
Comment (jdmcbr) [inline geopandas/base.py]: This TODO as well.

File: geopandas/base.py [hunk 2]
```diff
@@ -289,13 +316,13 @@ def geom_almost_equals(self, other, decimal=6):
             Decimal place presion used when testing for approximate equality.
         """
         # TODO: pass precision argument
```
Comment (jdmcbr) [inline geopandas/base.py]: I think I'm the one who missed this in a previous review, but that TODO can be removed now.

**Comment by jorisvandenbossche** [comment]
> I've never quite been able to figure out why base.py and geoseries.py are separate.

base.py contains things that are shared for both GeoSeries and GeoDataFrame. Not everything in GeoSeries should be inherited by GeoDataFrame, so it makes sense to have a base class for both I think.

**Comment by jorisvandenbossche** [comment]
Removed the todo's, thanks

**Comment by jorisvandenbossche** [comment]
Yeah, it is true that that might seem strange. But those methods that are shared between GeoSeries and GeoDataFrame (and thus are defined in base.py) do return a GeoSeries, so we need to import it somehow. Not sure if that can or should be done differently to make this easier to understand.

**Comment by jdmcbr** [comment]
I guess it felt to me like the advantage of having a separate module to hold the base methods used by both series and frames was somewhat outweighed by the necessity of frequent injection of geoseries into the base module. Basically, that in some sense that inclusion meant that the geoseries methods belonged in base, since many of the methods can't be defined without a geoseries. It isn't a big deal, by any means, but this PR brought that topic back to my mind.

**Pull request merged**